src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -375,20 +381,30 @@ private void handleOnDragDropped(TableRow<BibEntryTableViewModel> row, BibEntryT
             switch (ControlHelper.getDroppingMouseLocation(row, event)) {
                 case TOP, BOTTOM -> importHandler.importFilesInBackground(files).executeWith(Globals.TASK_EXECUTOR);
                 case CENTER -> {
-                    BibEntry entry = target.getEntry();
-                    switch (event.getTransferMode()) {
-                        case LINK -> {
-                            LOGGER.debug("Mode LINK"); // shift on win or no modifier
-                            importHandler.getLinker().addFilesToEntry(entry, files);
-                        }
-                        case MOVE -> {
-                            LOGGER.debug("Mode MOVE"); // alt on win
-                            importHandler.getLinker().moveFilesToFileDirAndAddToEntry(entry, files);
-                        }
-                        case COPY -> {
-                            LOGGER.debug("Mode Copy"); // ctrl on win
-                            importHandler.getLinker().copyFilesToFileDirAndAddToEntry(entry, files);
+                    try {
+                        BibEntry entry = target.getEntry();
+                        switch (event.getTransferMode()) {
+                            case LINK -> {
+                                LOGGER.debug("Mode LINK"); // shift on win or no modifier
+                                importHandler.getLinker().addFilesToEntry(entry, files);
+                            }
+                            case MOVE -> {
+                                LOGGER.debug("Mode MOVE"); // alt on win
+                                libraryTab.getIndexingTaskManager().setBlockingNewTasks(true);
```

Comment:
Is there a reason why this blocking shouldn't be added directly to the `importHandler` (or linker)?

Reply:
Not all users of importHandler have access to the IndexingTaskManager unfortunately.

Reply:
Mhh, do you have an example where this occurs? The whole linker/import handler stuff arose from the fact that there were many very similarly-looking code blocks scattered around the code base, that all handled imports in a similar but same way. Was a huge headache, so I would like not to introduce special handling in certain situations again.

Reply:
Oh, that sounds like a good reason to do it directly in the import handler.
The PreviewPanel was my concern, but I passed it along and made it work.